src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.macOS.cs

```diff
@@ -72,10 +72,11 @@ private static ILoaderPal FromBlob(ReadOnlySpan<byte> rawData, SafePasswordHandl
             bool ephemeralSpecified,
             SafeKeychainHandle keychain)
         {
-            ApplePkcs12Reader reader = new ApplePkcs12Reader(rawData);
+            ApplePkcs12Reader reader = new ApplePkcs12Reader();
 
             try
             {
+                reader.ParsePkcs12(rawData);
                 reader.Decrypt(password, ephemeralSpecified);
                 return new ApplePkcs12CertLoader(reader, keychain, password, exportable);
```

Comment:
I assume we're already confident (since this PR doesn't really change things in this regard) that everything up the chain is appropriately disposed of, e.g. that the call site to this method is appropriately disposing of the loader?

Reply:
Yeah, the only problem at hand was that the ctor throwing meant Dispose didn't get called.  Everything ultimately chains to a local catch+Dispose or returned and pinned into a using.